swe/tsunami/solver.py

```diff
@@ -15,23 +15,24 @@ def __init__(self, *args, extension=None, **kwargs):
         self.extension = extension
         super(TsunamiProblem, self).__init__(*args, **kwargs)
 
+
     def set_fields(self, adapted=False):
         self.fields = {}
         self.fields['viscosity'] = self.op.set_viscosity(self.P1)
         self.fields['diffusivity'] = self.op.set_diffusivity(self.P1)
-        # TODO: Should interpolate bathy from fresh
-        self.fields['bathymetry'] = self.op.set_bathymetry(self.P1, adapted=adapted)
```

Comment:
Again, why are you removing this? Please special case for your model by subclassing the swe / tsunami solver so that it sets fields differently.